packages/flutter_tools/lib/src/protocol_discovery.dart

```diff
@@ -22,9 +24,9 @@ class ProtocolDiscovery {
     assert(logReader != null);
     assert(portForwarder == null || defaultHostPort != null);
     _deviceLogSubscription = logReader.logLines.listen(_handleLine);
-    _timer = new Timer(const Duration(seconds: 60), () {
+    _timer = new Timer(kDiscoveryTimeout, () {
       _stopScrapingLogs();
-      _completer.completeError(new ToolExit('Timeout while attempting to retrieve URL for $serviceName'));
+      _completer.completeError(new ToolExit('Timeout attempting to discover $serviceName URL'));
```

Comment:
If there's no reason to believe the timeout will ever be hit except because things are slow, let's just remove the timeout.